src/Rings.jl

```diff
@@ -101,6 +101,13 @@ is_exact_type(R::Type{T}) where T <: RingElem = true
 # false unless explicitly specified
 is_domain_type(R::Type{T}) where T <: RingElem = false
 
+
+# Default implementation: covers all integral domains
+function is_nilpotent(a::T) where {T <: RingElem}
```

Comment:
This code is also valid for non-commutative rings, such as matrix rings, isn't it? if so let's generalize it:
```suggestion
function is_nilpotent(a::T) where {T <: NCRingElem}
```
Of course it then should rather go into `src/NCRings.jl`

Reply:
Done!